Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AtomRenderPlugin.cpp

```diff
@@ -127,11 +298,100 @@ namespace EMStudio
+            BuildMouseInteractionInternal(
+                AztfVi::MouseButtons(AztfVi::TranslateMouseButtons(QGuiApplication::mouseButtons())), keyboardModifiers,
+                BuildMousePick(m_animViewportWidget->mapFromGlobal(QCursor::pos()))));
+        debugDisplay->DepthTestOn();
+    }
+
+    bool AtomRenderPlugin::HandleMouseEvent(
+        const AzToolsFramework::ViewportInteraction::MouseInteractionEvent& mouseInteractionEvent)
+    {
+        if (!m_manipulatorManager)
+        {
+            return false;
+        }
+
+        using AzToolsFramework::ViewportInteraction::MouseEvent;
+        const auto& mouseInteraction = mouseInteractionEvent.m_mouseInteraction;
+
+        switch (mouseInteractionEvent.m_mouseEvent)
+        {
+        case MouseEvent::Down:
+            return m_manipulatorManager->ConsumeViewportMousePress(mouseInteraction);
+        case MouseEvent::DoubleClick:
+            return false;
+        case MouseEvent::Move:
+            {
+                const AzToolsFramework::ManipulatorManager::ConsumeMouseMoveResult mouseMoveResult =
+                    m_manipulatorManager->ConsumeViewportMouseMove(mouseInteraction);
+                return mouseMoveResult == AzToolsFramework::ManipulatorManager::ConsumeMouseMoveResult::Interacting;
+            }
+        case MouseEvent::Up:
+            return m_manipulatorManager->ConsumeViewportMouseRelease(mouseInteraction);
+        case MouseEvent::Wheel:
+            return m_manipulatorManager->ConsumeViewportMouseWheel(mouseInteraction);
+        default:
+            return false;
+        }
+    }
+
+    AzToolsFramework::ViewportInteraction::MousePick AtomRenderPlugin::BuildMousePick(const QPoint& point) const
+    {
+        AzToolsFramework::ViewportInteraction::MousePick mousePick;
+        mousePick.m_screenCoordinates = AzToolsFramework::ViewportInteraction::ScreenPointFromQPoint(point);
+        const auto [origin, direction] = m_animViewportWidget->ViewportScreenToWorldRay(mousePick.m_screenCoordinates);
+        mousePick.m_rayOrigin = origin;
+        mousePick.m_rayDirection = direction;
+        return mousePick;
+    }
+
+    AzToolsFramework::ViewportInteraction::MouseInteraction AtomRenderPlugin::BuildMouseInteractionInternal(
+        const AzToolsFramework::ViewportInteraction::MouseButtons buttons,
+        const AzToolsFramework::ViewportInteraction::KeyboardModifiers modifiers,
+        const AzToolsFramework::ViewportInteraction::MousePick& mousePick) const
+    {
+        AzToolsFramework::ViewportInteraction::MouseInteraction mouse;
+        mouse.m_interactionId.m_cameraId = AZ::EntityId(AZ::EntityId::InvalidEntityId);
+        mouse.m_interactionId.m_viewportId = m_animViewportWidget->GetViewportContext()->GetId();
+        mouse.m_mouseButtons = buttons;
+        mouse.m_mousePick = mousePick;
+        mouse.m_keyboardModifiers = modifiers;
+        return mouse;
+    }
```

Comment:
suggestion: I think there's now come util code in `ViewportTypes.h/cpp` that might be able to be used to avoid some of this boilerplate (worth checking in case)